modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -297,6 +298,8 @@ object Inputs {
       else if os.isDir(path) then Right(Seq(Directory(path)))
       else if acceptFds && arg.startsWith("/dev/fd/") then
         Right(Seq(VirtualScript(content, arg, os.sub / s"input-${idx + 1}.sc")))
+      else if isRunWithShebang && String(content).startsWith("#!") then
```

Comment:
we have a regex for shebang headers defined at `scala.build.preprocessing.SheBang.sheBangRegex`, we could check against it as well